lib/Ast_to_Mir.ml

```diff
@@ -196,18 +196,6 @@ let rec trans_checks cvarname ctype t =
   | Ast.Covariance -> [Check {check with cfname= "cov_matrix"}]
   | Ast.Offset _ | Ast.Multiplier _ | Ast.OffsetMultiplier (_, _) -> []
 
-(** Adds Mir statements that validate the variable once it has been read.
-    The code to read it in is emitted in the backend.contents
-    Here we intentionally only check declarations at the top level, i.e.
-    we only recurse into blocks and lists as we don't care about other declarations.
-*)
-let tvdecl_checks {tvident; tvtrans; tvtype; tvloc} =
-  let with_sloc stmt = {sloc= tvloc; stmt} in
-  let check_stmts =
-    List.map ~f:with_sloc (trans_checks tvident tvtype tvtrans)
-  in
-  with_sloc (SList check_stmts)
-
 let trans_tvdecl {Ast.stmt_typed; stmt_typed_loc; _} =
   match stmt_typed with
   | Ast.VarDecl
```

Comment:
Presumably, we will throw an error later if we find a `None` for the `Mir.tvdecl option`?

Reply:
It's just used to filter below: 
```
 List.(filter_opt (map ~f:trans_tvdecl lst))
```

Reply:
That's fine.